app/models/user.rb

```diff
@@ -115,7 +115,7 @@ def self.new_with_session(params, session)
   end
 
   def use_default_photo_pic?
-    self.profile_photo_url ==
+    read_attribute(:profile_photo_url) ==
```

Comment:
Because I override self.profile_photo_url calls my override function,  while read_attribute returns the value in db.

Same as our course logo implementation:
https://github.com/Coursemology/coursemology.org/blob/development/app/models/course.rb#L420